package idl;
enum FeedOption {
    // common
    CONNECT = 0;
    DISCONNECT = 1;
    MESSAGE = 2;

    PIPE = 3; // ininer->outer
    ACK = 4;  // outer->inner
    NEW = 5; // client->outer
}

message FeedRemoteInfo {
    required int32 port = 1;
    optional string ip = 2;
}

message FeedAction {
    required FeedOption option = 1;
    optional int32 fd = 2;
    optional bytes data = 3;
    optional FeedRemoteInfo remoteInfo = 4;
}